shared/fs/common/path-item-action-container.js

```diff
@@ -53,62 +50,127 @@ const mapDispatchToProps = (dispatch: Dispatch, {path}: OwnProps) => ({
     : {}),
 })
 
-const getRootMenuItems = (stateProps, dispatchProps, path: Types.Path) => {
-  const {_pathItems, fileUIEnabled, _username} = stateProps
-  const {_showInFileUI, _saveMedia, _shareNative, _download, _ignoreFolder, copyPath} = dispatchProps
-  const pathItem = _pathItems.get(path, Constants.unknownPathItem)
-  let menuItems = []
-
-  !isMobile &&
-    fileUIEnabled &&
-    menuItems.push({
-      title: 'Show in ' + fileUIName,
-      onClick: () => _showInFileUI(),
-    })
-
-  isMobile &&
-    pathItem.type !== 'folder' &&
-    Constants.isMedia(pathItem) &&
-    menuItems.push({
-      title: 'Save',
-      onClick: () => _saveMedia(),
-    })
-
-  const shouldShowShareNative = isMobile && pathItem.type === 'file'
-  shouldShowShareNative &&
-    menuItems.push({
-      title: 'Send to other app',
-      onClick: () => _shareNative(),
-    })
-
-  const shouldDownload = (isAndroid && pathItem.type !== 'folder') || !isMobile
-  shouldDownload &&
-    menuItems.push({
-      title: 'Download a copy',
-      onClick: () => _download(),
-    })
-
-  Constants.showIgnoreFolder(path, pathItem, _username) &&
-    menuItems.push({
-      title: 'Ignore this folder',
-      onClick: () => _ignoreFolder(),
-      subTitle: 'The folder will no longer appear in your folders list.',
-      danger: true,
-    })
-
-  menuItems.push({
+type MenuItemAppender = (
```

Comment:
I think its likely better for this construct to change. I was speaking to danny about this yesterday. I think it likely makes more sense for the mapStateToDispatch to just create the dispatchProps and the component itself to make these menu items. 

My argument is the role of the connector is to bridge the simple props the component needs to draw itself and to give it callbacks it can call to do stuff. The data structure MenuItem is an implementation detail of the component and doesn't really have anything to do with redux.

I argue better to just be like 
```
const mergeProps...
return {
  showInFileUI: !isMobile && stateProps._fileUIEnabled && dispatchProps._showInFileUI
...
}
```

then the component itself does in its render

```
const menuItems = [
 ...(this.props.showInFileUI ? {onClick: this.props.showInFileUI, label: 'Show in Findere'} : {}),
]
```

For example if we change how menu works or if the view changes to be a button bar instead of a menu the connector would have to change, which I argue isn't ideal

Reply:
Sounds good; will make the change. Thanks for explaining the details!